src/utilities/EventEmitter.test.ts

```diff
@@ -0,0 +1,242 @@
+      expect(eventEmitter.events['myOtherEvent']).not.toContain(listener);
+      expect(eventEmitter.events['myThirdEvent']).toContain(listener);
+    });
+
+    it('Removes all listeners from an event', () => {
+      eventEmitter.off('myEvent');
+      expect(eventEmitter.events['myEvent']).toBe(undefined);
+      expect(eventEmitter.events['myOtherEvent']).toContain(listener);
+    });
+
+    it('Removes all listeners from multiple events', () => {
+      eventEmitter.off(['myEvent', 'myOtherEvent']);
+      expect(eventEmitter.events['myEvent']).toBe(undefined);
+      expect(eventEmitter.events['myOtherEvent']).toBe(undefined);
+      expect(eventEmitter.events['myThirdEvent']).toContain(listener);
+    });
+
+  });
+
+  describe('calling the listeners method', () => {
+    beforeEach(() => {
+      eventEmitter = new EventEmitter();
+      eventEmitter.on(listener);
+      eventEmitter.on(altListener);
+      eventEmitter.on('myEvent', listener);
+      eventEmitter.once(listener);
+      eventEmitter.once(altListener);
+      eventEmitter.once('myEvent', listener);
+      eventEmitter.once('myEvent', altListener);
+    });
+
+    it('Returns all listeners for a given event', () => {
+      expect(eventEmitter.listeners('myEvent')).toHaveLength(3);
+      expect(eventEmitter.listeners('myOtherEvent')).toBe(null);
+      eventEmitter.on('myEvent', listener);
+      expect(eventEmitter.listeners('myEvent')).toHaveLength(4);
+      eventEmitter.once('myEvent', listener);
+      expect(eventEmitter.listeners('myEvent')).toHaveLength(5);
+    });
+
+  });
+
+  describe('calling the once method', () => {
+    beforeAll(() => {
+      eventEmitter = new EventEmitter();
+    });
+
+    it('Adds a listener to anyOnce on calling `once` with a listener', () => {
+      eventEmitter.once(listener);
+      expect(eventEmitter.anyOnce).toHaveLength(1);
+    });
+
+    it('Adds a listener to an eventOnce on calling `once` with a listener and event name', () => {
+      eventEmitter.once('myEvent', listener);
+      expect(eventEmitter.eventsOnce['myEvent']).toHaveLength(1);
+    });
+
+    it('Adds a listener to multiple eventOnce fields on calling `once` with a listener and event name; And after emitting any of the events, all are removed', () => {
+      eventEmitter.once(['myEvent', 'myOtherEvent', 'myThirdEvent'], listener);
+      expect(eventEmitter.eventsOnce['myEvent']).toHaveLength(1);
+      expect(eventEmitter.eventsOnce['myOtherEvent']).toHaveLength(1);
+      expect(eventEmitter.eventsOnce['myThirdEvent']).toHaveLength(1);
+      expect(eventEmitter.emit('myEvent'));
+      expect(eventEmitter.eventsOnce['myEvent']).toBe(undefined);
+      expect(eventEmitter.eventsOnce['myOtherEvent']).toBe(undefined);
+      expect(eventEmitter.eventsOnce['myThirdEvent']).toBe(undefined);
+    });
+  });
+
+  describe('calling the emit method', () => {
+    beforeEach(() => {
+      eventEmitter = new EventEmitter();
+      eventEmitter.once(listener);
```

Comment:
It probably would fine to just define the `listener` here, instead of 200 lines above to improve readability – WDYT?

Reply:
I can define in `beforeEach`, it doesn't bother me